### PR TITLE
Update the CMake file to add more checks

### PR DESCRIPTION
**Pull request by AniruddhaKanhere**
Description
-----------
This PR updates the CMake file used for unit-testing to add checks for missing utilities and process to acquire the said missing utilities.

Test Steps
-----------

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/unit-test/CMakeLists.txt [hunk 1]
```diff
@@ -30,11 +30,67 @@ set( KERNEL_DIRECTORY ${MODULE_ROOT_DIR}/test/FreeRTOS-Kernel )
...
+                     WORKING_DIRECTORY ${MODULE_ROOT_DIR} )
+endif()
+
+# Find 'unifdef' utility.
+find_program( UNIFDEF_found unifdef )
+
+if( NOT UNIFDEF_found )
+    # unifdef was not found, install it and inform the user of the location of installation.
+    message( WARNING "unifdef utility not found. Installing it." )
+    execute_process( COMMAND apt-get install unifdef )
```
Comment (alfred2g) [inline test/unit-test/CMakeLists.txt:64]: this might be problematic on redhat/rpm based systems

File: test/unit-test/CMakeLists.txt [hunk 2]
```diff
@@ -30,11 +30,67 @@ set( KERNEL_DIRECTORY ${MODULE_ROOT_DIR}/test/FreeRTOS-Kernel )
...
+    message( STATUS "lcov installed at location ${LCOV_found}" )
+endif()
+
+# Find whether ruby is installed.
+find_program( RUBY_found ruby )
+
+if( NOT RUBY_found )
+    # Ruby was not found, install it and inform the user of the location of installation.
+    message( WARNING "Ruby (required by CMock) not found. Installing it." )
+    execute_process( COMMAND apt-get install ruby )
```
Comment (alfred2g) [inline test/unit-test/CMakeLists.txt:88]: same here

**Review by alfred2g — APPROVED** [review]

File: test/unit-test/CMakeLists.txt [hunk 3]
```diff
@@ -30,11 +38,126 @@ set( KERNEL_DIRECTORY ${MODULE_ROOT_DIR}/test/FreeRTOS-Kernel )
...
+                                 "Otherwise, you can set the package manager and the install command. Example for Red Hat Enterprise Linux:\n"
+                                 "    cmake -S. -B build <Additional commands/options> -DSYSTEM_PKG_MNGR=yum -DSYSTEM_INSTALL_CMD=install" )
+        endif()
+    endif()
+endif()
+
+# Find 'unifdef' utility.
+find_program( UNIFDEF_found unifdef )
+
+if( NOT UNIFDEF_found )
```
Comment (aggarg) [inline test/unit-test/CMakeLists.txt:97]: This seems like a repeated pattern for checking and installing tool. Should we write a function?

Comment (AniruddhaKanhere) [reply test/unit-test/CMakeLists.txt]: Yes, I agree. I made the change in the commit https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/404/commits/b16fe1172e251f02eb0ba9655f48987f346f84be

**Review by aggarg — APPROVED** [review]

**Review by aggarg — APPROVED** [review]

**Pull request merged**